### PR TITLE
CY-2593 Add support for running py3

### PR DESCRIPTION
**Pull request by mabuaisha**

### CODE REVIEW

File: fabric_plugin/tasks.py [hunk 1]
```diff
@@ -70,6 +70,25 @@ def _load_private_key(key_contents):
     )
 
 
+def _check_python_version(run_func):
+    def _which_python():
```
Comment (tehasdf) [inline fabric_plugin/tasks.py:74]: nesting the function serves no purpose?

File: fabric_plugin/tasks.py [hunk 2]
```diff
@@ -273,6 +292,10 @@ def upload_script(self, script_path):
             remote_path_suffix)
         self._sftp.put(script_path, self.remote_script_path)
 
+    @staticmethod
+    def is_py_script(script_path):
+        return True if script_path.endswith('.py') else False
```
Comment (tehasdf) [inline fabric_plugin/tasks.py:297]: this is just `return script_path.endswith('.py')`

File: fabric_plugin/tasks.py [hunk 3]
```diff
@@ -368,9 +399,12 @@ def run_script(ctx,
             env[CTX_SOCKET_URL] = proxy.socket_url
             env['LOCAL_{0}'.format(CTX_SOCKET_URL)] = proxy.socket_url
             files.upload_env_script(env)
-            command = 'cd {0} && source {1} && {2}'.format(
-                cwd, files.remote_env_script_path, command)
-            run = conn.sudo if use_sudo else conn.run
+            if python_prefix:
+                command = 'cd {0} && source {1} && {2} {3}'.format(
+                    cwd, files.remote_env_script_path, python_prefix, command)
```
Comment (tehasdf) [inline fabric_plugin/tasks.py:404]: there is already a `command_prefix` in this function
we should use that instead, ie. if it's not set, only then figure out the python executable automatically

File: fabric_plugin/tasks.py [hunk 4]
```diff
@@ -70,6 +70,25 @@ def _load_private_key(key_contents):
...
+        try:
+            # This is usually reference to python2 and it could reference to
+            # python3 if `alternatives mechanism used to enable the
+            # unversioned python`
+            py_version = run_func(
+                'python -c \'import sys; print(sys.version_info[0])\''
+            ).stdout
+        except Exception:
+            py_version = '3'
+            py_path = run_func('which python3').stdout.rstrip()
```
Comment (tehasdf) [inline fabric_plugin/tasks.py:84]: you're already doing `which` so what's the reason to have the `python -c` too?

File: fabric_plugin/tasks.py [hunk 5]
```diff
@@ -351,9 +374,17 @@ def run_script(ctx,
...
             _RemoteFiles(conn, process.get('base_dir')) as files:
+        run = conn.sudo if use_sudo else conn.run
         files.upload_script(local_script_path)
         fabric_ctx = _FabricCtx(ctx, files)
 
+        # handle execution python scripts
+        python_prefix = None
+        if files.is_py_script(local_script_path):
+            py_version, py_path = _check_python_version(run)
+            if py_version == '3':
```
Comment (tehasdf) [inline fabric_plugin/tasks.py:385]: not sure why 2 and 3 are different here, instead of just using the path

File: fabric_plugin/tasks.py (hunk 1 above)
Comment (mabuaisha) [reply fabric_plugin/tasks.py]: Done

File: fabric_plugin/tasks.py (hunk 2 above)
Comment (mabuaisha) [reply fabric_plugin/tasks.py]: Done

File: fabric_plugin/tasks.py (hunk 3 above)
Comment (mabuaisha) [reply fabric_plugin/tasks.py]: Done

File: fabric_plugin/tasks.py (hunk 4 above)
Comment (mabuaisha) [reply fabric_plugin/tasks.py]: Remove the whole method, I'm depending on source right now

File: fabric_plugin/tasks.py (hunk 5 above)
Comment (mabuaisha) [reply fabric_plugin/tasks.py]: Removed the whole method

**Comment by mabuaisha** [comment]
@tehasdf I updated the code, please let me know what do you think and if there are more notes to add

**Pull request merged**